source/whats-new.txt

```diff
@@ -67,6 +67,21 @@ The 5.0 driver release introduces the following features:
 - Changes to the ``ConnectionId`` class. To learn more, see the
   :ref:`Version 5.0 Breaking Changes <java-breaking-changes-v5.0>`.
 
+- A new ``authorizedCollections`` option has been added to the
+  ``listCollections`` command. It can be configured using the
+  ``MongoDatabase.listCollectionNames()`` method. This introduces a
+  :ref:`breaking change to the listCollections
+  
+- The ``listCollections`` command has an ``authorizedCollections`` option. It can be configured using the
+  ``MongoDatabase.listCollectionNames()`` method. This introduces a :ref:`binary
+  breaking change to the listCollections
+  command<java-breaking-changes-v5.0-list-collections>`.
```

Comment:
The concept of binary compatibility is not applicable to the `listCollections` MongoDB command, it is in the current context only applicable to the methods `MongoDatabase.listCollectionNames`.

I should have probably wrote a more detailed "Documentation Changes Summary" in https://jira.mongodb.org/browse/JAVA-4353. Apologies, let me fix that mistake here.

The methods `com.mongodb.client.MongoDatabase.listCollectionNames` were returning `MongoIterable<String>`, which did not allow configuring the `authorizedCollections` option of the `listCollections` command. To support this option, the return type of the methods was changed to `com.mongodb.client.ListCollectionNamesIterable`, which broke binary compatibility. Now users may use the method `ListCollectionNamesIterable.authorizedCollections` to configure the `authorizedCollections` option.

A similar change was done to the `listCollectionNames` methods in the following classes/interfaces:

- `com.mongodb.reactivestreams.client.MongoDatabase`
- `org.mongodb.scala.MongoDatabase`
- `com.mongodb.kotlin.client.MongoDatabase`
- `com.mongodb.kotlin.client.coroutine.MongoDatabase`

However, the original and new return types of all the `listCollectionNames` methods are different across all the aforementioned classes/interfaces, which is why I originally formulated the "Documentation Changes Summary" the way I did: it avoids referring to those types by names, while still being correct (yet apparently not clear enough).